opencti-platform/opencti-graphql/src/graphql/sseMiddleware.js

```diff
@@ -350,10 +355,18 @@ const createSseMiddleware = () => {
     const refs = stixRefsExtractor(stixData, generateStandardId);
     const missingElements = await resolveMissingReferences(context, req, refs, cache);
     const missingInstances = await storeLoadByIdsWithRefs(context, req.user, missingElements);
-    const missingAllPerIds = missingInstances.map((m) => {
-      return [m.internal_id, m.standard_id, ...(m.x_opencti_stix_ids ?? [])].map((id) => ({ id, value: m }));
-    }).flat();
+    const missingAllPerIds = missingInstances.map((m) => [m.internal_id, m.standard_id, ...(m.x_opencti_stix_ids ?? [])].map((id) => ({ id, value: m }))).flat();
     const missingMap = new Map(missingAllPerIds.map((m) => [m.id, m.value]));
+    if (stixData.type === STIX_TYPE_RELATION || stixData.type === STIX_TYPE_SIGHTING) {
+      // Check for a relation that the from and the to is correctly accessible.
+      const fromId = stixData.source_ref ?? stixData.sighting_of_ref;
+      const toId = stixData.target_ref ?? stixData.where_sighted_refs[0];
+      const hasFrom = missingMap.has(fromId) || cache.has(fromId);
+      const hasTo = missingMap.has(toId) || cache.has(toId);
+      if (!hasFrom || !hasTo) {
```

Comment:
I am not sur this is exactly what we want
If there are in a platform
- a sector : marking RED
- a sector : marking GREEN
- a relationship between this two sectors : marking GREEN
- user A : marking GREEN

If user A modify the relationship, the modification is not displayed in the stream

Reply:
Its exactly what is expected. If both part of the relation is not accessible the relation must not be published.